File: day1/script2.py
import os 

def read_input():
    path = os.path.dirname(os.path.realpath(__file__))
    data = open(f'{path}/input.txt', 'r').read().strip().split('\n')
    return data

data = read_input()
loc1, loc2 = zip(*list(map(lambda line: [int(e) for e in line.split()], data)))

print(sum([num * loc2.count(num) for num in loc1]))

File: day1/script2.r

read_input <- function() {
  data <- read.table("day1/input.txt", header = F, sep = "")
  return(data)
}
data <- read_input()
data$V1 <- sort(data$V1, decreasing = F)
data$V2 <- sort(data$V2, decreasing = F)
print(sum(sapply(data$V1, \(x) x * length(data$V2[data$V2 == x]))))